### PR TITLE
feat: TypeScript SDK — Next.js error capture package

### PR DESCRIPTION
**Pull request by moomooskycow**
## Summary

- Replaces single-file `canary.ts` client with `@canary-obs/sdk` npm package
- Provides `initCanary()` / `captureException()` / `captureMessage()` public API
- Next.js integration via `onRequestError` hook for `instrumentation.ts`
- PII scrubbing (email redaction + custom patterns) before send
- Bounded retry queue (1 retry, 2s timeout, max 10 inflight), zero runtime deps
- Dual ESM/CJS build with TypeScript declarations

## Test plan

- [x] 23 unit tests covering scrub, client transport, public API, and Next.js hook
- [x] `npm pack --dry-run` produces valid package with all entry points
- [x] Build generates ESM, CJS, and `.d.ts` outputs
- [ ] Manual: install in a Next.js app and verify error capture end-to-end

## What Changed

```mermaid
flowchart LR
    subgraph Before
        A[canary.ts] -->|class Canary.capture| B[POST /api/v1/errors]
    end
```

```mermaid
flowchart LR
    subgraph After
        C[instrumentation.ts] -->|onRequestError| D["@canary-obs/sdk"]
        E[error.tsx] -->|captureException| D
        F[catch blocks] -->|captureException| D
        D -->|scrub PII| G[client.ts queue]
        G -->|fetch + retry| H[POST /api/v1/errors]
    end
```

The new shape separates concerns (scrub → queue → transport) and adds Next.js-native hooks instead of requiring manual wiring.

## Acceptance Criteria

From #26:

- [x] `initCanary({ endpoint, apiKey, service })` captures server errors via `onRequestError`
- [x] `captureException(error)` sends POST with error_class, message, stack_trace
- [x] Fails silently when endpoint is unreachable (no throw)
- [x] `npm pack` produces valid package with ESM and CJS exports
- [x] PII scrubbing redacts email addresses before sending

## Test Coverage

- `test/scrub.test.ts` — 8 tests: email redaction, custom patterns, edge cases
- `test/client.test.ts` — 6 tests: auth header, retry, queue cap, error resilience
- `test/index.test.ts` — 7 tests: captureException, captureMessage, PII scrubbing, silent failure
- `test/nextjs.test.ts` — 2 tests: onRequestError context propagation

## Merge Confidence

**High** — zero runtime deps, 23 passing tests, clean build. Residual risk: no end-to-end test against live Canary instance (requires integration test in consuming app).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Introduced a TypeScript error-tracking SDK with init, exception and message capture APIs.
  * Added PII scrubbing (string and object-level) to redact sensitive data.
  * Added Next.js request error instrumentation and a client-side/provider initialization pattern.
  * Implemented retry logic and queue management for error submissions.

* **Documentation**
  * Revised integration guide with updated initialization and verification steps.

* **Tests**
  * Added comprehensive tests covering capture, retries, queueing, Next.js hooks, and scrubbing.

* **Chores**
  * Added package, build, TypeScript project config and gitignore.

### CODE REVIEW

File: clients/typescript/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  const { errorClass, message, stackTrace } = normalizeError(error);
+
+  return client.send({
+    error_class: errorClass,
+    message: scrubPii ? scrub(message, scrubRules)! : message,
+    severity: opts.severity ?? "error",
+    stack_trace: scrubPii ? scrub(stackTrace, scrubRules) : stackTrace,
+    context: opts.context,
+    fingerprint: opts.fingerprint,
+  });
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply clients/typescript/src/index.ts]: Fixed in 50856f7. Added `scrubObject()` that recursively walks context objects/arrays and scrubs all string values. Both `captureException` and `captureMessage` now pass `context` through `scrubObject()` when `scrubPii` is enabled. Test added to verify.
